src/ir/LocalGraph.cpp

```diff
@@ -143,7 +144,7 @@ struct Flower : public CFGWalker<Flower, Visitor<Flower>, Info> {
         auto& gets = allGets[index];
         if (gets.empty()) continue;
         work.push_back(block.get());
-        seen.clear();
```

Comment:
I think we need to initialize all blocks's `lastSeenIndex` here (as opposed to just once when creating the block): we can reach this location for the same index multiple times, and we need to forget previous settings of `lastSeenIndex` each time we start here (so it is logically the same as `seen.clear()` from before).

That seems like it would add some overhead, we'd need to measure that.

Also, given that, I don't think we need to store the index - just a boolean, `seen`, seems enough. In other words, this would move the `seen` map into a `seen` property on each block.